docker/dev/build-dev.sh

```diff
@@ -1,20 +0,0 @@
-#!/bin/bash
```

Comment:
After we split the Dockerfile based on different use cases, we no longer need these scripts to manage build context.